pyro/contrib/autoguide/advi.py

```diff
@@ -28,23 +28,59 @@ def _product(shape):
     return result
 
 
-class ADVIMaster(object):
+class AutoGuide(object):
     """
-    Container class to combine multiple ADVI strategies.
+    Base class for automatic guides.
+
+    Derived classes must implement the :meth:`__call__` method.
+
+    Auto guides can be used individually or combined in an
+    :class:`AutoGuideList` object.
+    """
+    def __init__(self, model):
+        self.master = None
+        self.model = model
+        self.prototype_trace = None
+
+    def __call__(self, *args, **kwargs):
+        """
+        A guide with the same ``*args, **kwargs`` as the base ``model``.
+
+        :return: A dict mapping sample site name to sampled value.
+        :rtype: dict
```

Comment:
Are you using this return value anywhere?

Reply:
Yes, Slawek uses it to inspect latent variables. I guess we could force users to trace the guide instead and eliminate this logic in a future PR.